test_runner/regress/test_compute_metrics.py

```diff
@@ -1,21 +1,146 @@
-from fixtures.neon_fixtures import NeonEnv
+from __future__ import annotations
 
+import shutil
+import subprocess
+import time
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, TypedDict, cast
 
-def test_compute_metrics(neon_simple_env: NeonEnv):
+import pytest
+import requests
+import yaml
+from fixtures.log_helper import log
+
+if TYPE_CHECKING:
+    from fixtures.neon_fixtures import NeonEnv
+    from pytest import MonkeyPatch
+
+    class Metric(TypedDict):
+        metric_name: str
+        type: str
+        help: str
+        key_labels: tuple[str] | None
+        values: tuple[str]
+        query: str | None
+        query_ref: str | None
+
+    class Collector(TypedDict):
+        collector_name: str
+        metrics: list[Metric]
+        queries: list[Query] | None
+
+    class Query(TypedDict):
+        query_name: str
+        query: str
+
+
+CONFIG_DIR = Path(__file__).parents[2] / "compute" / "etc"
+
+
+@pytest.mark.parametrize(
+    "collector_config", ["neon_collector.yml", "neon_collector_autoscaling.yml"]
+)
+def test_sql_exporter_metrics_smoke(neon_simple_env: NeonEnv, collector_config: Path):
+    """
+    This is a smoke test to ensure the metrics SQL queries sql_exporter work
+    without errors.
+    """
+    env = neon_simple_env
+
+    endpoint = env.endpoints.create("main")
+    endpoint.respec(skip_pg_catalog_updates=False)
+    endpoint.start()
+
+    # Extract all the SQL queries from the sql_exporter config files, and run
+    # them. We don't know what they're supposed to return, so all we can test
+    # here is that they run without error.
+    with open(CONFIG_DIR / collector_config, "r", encoding="utf-8") as f:
+        collector = cast("Collector", yaml.safe_load(f))
+
+        for metric in collector["metrics"]:
+            query = metric.get("query")
+            if query is not None:
+                log.info(
+                    "Checking query for metric %s in %s", metric["metric_name"], collector_config
+                )
+                endpoint.safe_psql(query)
+
+        queries = collector.get("queries")
+        if queries is not None:
+            for query in queries:
+                log.info("Checking query %s in %s", query["query_name"], collector_config)
+                endpoint.safe_psql(query["query"])
+
+    endpoint.stop_and_destroy()
+
+
+SQL_EXPORTER = shutil.which("sql_exporter")
+
+
+@pytest.mark.skipif(SQL_EXPORTER is None, reason="sql_exporter is not in the PATH")
```

Comment:
It silently skips tests if `sql_exporter` is not installed. 
I don't think this is a desired behaviour.

Reply:
Not everyone is going to have sql_exporter installed on their system. I don't think that should be a requirement. Building it is not that easy.

Reply:
Agreed.

I think the danger here is that we'll change the CI setup and forget to install sql_exporter, and then all the sql_exporter tests are silently skipped without anyone noticing. Having an explicit flag to enable/disable the tests, and fail if sql_exporter is not installed, would alleviate that.

I'm not sure how we've done that kind of conditional tests in the past..

Reply:
I am going to make it not skip the tests if the CI env var is set.

Reply:
Actually, I'll add support for [testcontainers](https://testcontainers.com/guides/getting-started-with-testcontainers-for-python/), so the e2e test will always run. We can use the same sql_exporter [image](https://hub.docker.com/r/burningalchemist/sql_exporter) that we use in production.